src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || size < index) {
+            throw new IndexOutOfBoundsException("The index " + index + " is invalid "
+                    + "the actual size of array is " + size);
+        }
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else if (index == 0) {
+            newNode.next = first;
+            first = newNode;
+        } else if (index == size) {
+            last.next = newNode;
+            last = newNode;
+        } else if (index > 0 && size > index) {
+            Node<T> current = first;
+            int c = 0;
+            while (current != null && c != index) {
+                current = current.next;
+                c++;
+            }
+            current.prev.next = newNode;
+            newNode.prev = current.prev;
+            current.prev = newNode;
+            newNode.next = current;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        Node<T> newNode;
+        for (T t : list) {
+            newNode = new Node<>(t);
+            add(newNode.item);
+        }
```

Comment:
Don't create repeating code. You're repeating the logic of `add(T value)` in `addAll(List<T> list)`. You should directly call `add(value)` instead of creating a new node and then calling `add(newNode.item)`. This will avoid code duplication and ensure that any logic for adding a single element is centralized.